server/src/projects/projects.service.ts

```diff
@@ -46,6 +46,40 @@ export class ProjectsService {
     }
   }
 
+  public async getProject(projectId: string, contactId: string) {
+    try {
+      const { records } = await this.crmService.get('dcp_projects', `
+        $filter=
+          dcp_dcp_project_dcp_projectapplicant_Project/
+            any(o:
+              o/_dcp_applicant_customer_value eq '${contactId}'
+              and o/statuscode eq ${APPLICANT_ACTIVE_STATUS_CODE}
+            )
+          and (
+            dcp_visibility eq ${PROJECT_VISIBILITY_APPLICANT_ONLY}
+            or dcp_visibility eq ${PROJECT_VISIBILITY_GENERAL_PUBLIC}
+          )
+          and statecode eq ${PROJECT_ACTIVE_STATE_CODE}
+          and dcp_name eq '${projectId}'
+        &$expand=
+          dcp_dcp_project_dcp_package_project,
+          dcp_dcp_project_dcp_projectapplicant_Project(
+            $filter= statuscode eq ${APPLICANT_ACTIVE_STATUS_CODE}
+          )
+      `);
```

Comment:
The client only passes `dcp_name` to the backend, though. How would we get `dcp_projectid`, without doing a separate lookup using `dcp_name`? 

I say the client only passes `dcp_name` to the backend assuming a few things:
  1. The client URLs are designed similar to ZAP search, where the URL for an individual project page is something like `/projects/2020MK2`
  2. If a user visits the individual project page directly, the route won't preload all the proejcts. It will only send one request to the backend for the project specified in the URL param. 

I guess we could modify the route to preload all projects (hit the /projects endpoint) first. Thoughts?

Reply:
Why does it only pass `dcp_name` to the backend, though? I'm not sure it should be doing that anyway.

> How would we get `dcp_projectid`, without doing a separate lookup using `dcp_name`?
Not sure I follow — the client just needs to update the `link-to`s to use the correct id.

> 1. The client URLs are designed similar to ZAP search, where the URL for an individual project page is something like `/projects/2020MK2`

Doesn't mean we should be continue doing it that way. If we need this for vanity reasons, we should design it that way intentionally.


> 2\. If a user visits the individual project page directly, the route won't preload all the projects (like by hitting the `/projects` endpoint). It will only send one request to the backend for the project specified in the URL param.

Exactly — that's what projects/id should do. Why does a single project page need all the projects?

Is there a wireframe for this?

This implementation-specific issue style is a little too presumptuous....

Reply:
I assumed that planners/applicants would want that predictable human-understandable ulurp number in the URL. But you're right, we don't need to stick with that.

On a separate note, Ryan mentioned we shouldn't be using CRM record ids in the url. Though we can address this in a separate issue.

The single project page imo shouldn't load all projects. I was thinking through a workaround for only passing dcp_name

I'll change it to use projectid in the URL, and let others know of the change

Reply:
> Ryan mentioned we shouldn't be using CRM record ids in the url

Really? Ack. Why? We do that _everywhere_. Lemme know if there's an issue created...

> I assumed that planners/applicants would want that predictable human-understandable ulurp number in the URL. But you're right, we don't need to stick with that ... I'll change it to use projectid in the URL, and let others know of the change

Thanks... I'm pushing for consistency here because it's how our other routes are (mostly) designed. If we need to move away from that, let's make an issue of it and do it. Vanity URLs are a totally valid request, but let's not implement it on accident.